src/Common/src/TypeSystem/Ecma/EcmaMethod.cs

```diff
@@ -415,5 +415,27 @@ public override PInvokeMetadata GetPInvokeMethodMetadata()
 
             return new PInvokeMetadata(moduleName, name, (PInvokeAttributes)import.Attributes);
         }
+
+        public override PInvokeArgumentMetadata[] GetPInvokeArgumentMetadata()
+        {
+            if (!IsPInvoke)
+                return default(PInvokeArgumentMetadata[]);
+
+            MetadataReader metadataReader = MetadataReader;
+
+            ParameterHandleCollection parameterHandles = metadataReader.GetMethodDefinition(_handle).GetParameters();
+            PInvokeArgumentMetadata[] argumentMetadata = new PInvokeArgumentMetadata[parameterHandles.Count];
```

Comment:
What's our general policy for caching arrays like this? Is it usually that we only traverse all the methods exactly once, or should we simply create a new PInvokeEcmaMethod instead that has the right fields?